data/metric/metrics.go

```diff
@@ -163,6 +164,12 @@ func (ms *Set) elapsedDifference(name string, absolute interface{}, sourceType S
 	}
 
 	elapsed = newValue - oldValue
+
+	if elapsed < 0 && sourceType.IsPositive() {
```

Comment:
this is just a suggestion. Instead of having a `IsPositive` method that returns elapsed value and it validates depending the source, something like 
```
elapsed, err := sourceType.Elapsed(newValue, oldValue)

func (t SourceType) Elapsed(new, old int) (int, error) {
    e := n -old
    if e >= 0 {
        return e, nil
    }
    var err error
    if t. PRATE || t. PDELTA {
       err = ErrNegativeDiff
   }
   return e, err
}

```
In this case we delegate the responsability of validation to the source type and we don't have to check this in the `SetMetric` method.

Once again I like the current approach just to show how would be in this other way